pkg/sources/elasticsearch/api.go

```diff
@@ -0,0 +1,471 @@
+package elasticsearch
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"slices"
+	"strings"
+	"sync"
+	"time"
+
+	es "github.com/elastic/go-elasticsearch/v8"
+	"github.com/elastic/go-elasticsearch/v8/esapi"
+)
+
+const PAGE_SIZE = 10
+
+type IndexStatus int
+
+type FilterParams struct {
+	indexPattern   string
+	queryJSON      string
+	sinceTimestamp string
+}
+
+type PointInTime struct {
+	ID        string `json:"id"`
+	KeepAlive string `json:"keep_alive"`
+}
+
+type SearchRequestBody struct {
+	PIT         PointInTime    `json:"pit"`
+	Sort        []string       `json:"sort"`
+	SearchAfter []int          `json:"search_after,omitempty"`
+	Query       map[string]any `json:"query,omitempty"`
+}
+
+type Document struct {
+	id        string
+	timestamp string
+	message   string
+}
+
+type Index struct {
+	name                   string
+	documentCount          int
+	latestTimestamp        time.Time
+	latestTimestampLastRun time.Time
+	latestDocumentIDs      []string
+	lock                   sync.RWMutex
+}
+
+type Indices struct {
+	indices                 []*Index
+	documentCount           int
+	processedDocumentsCount int
+	filterParams            *FilterParams
+	lock                    sync.RWMutex
+}
+
+type elasticSearchRequest interface {
+	Do(providedCtx context.Context, transport esapi.Transport) (*esapi.Response, error)
+}
+
+func (fp *FilterParams) Query(latestTimestamp time.Time) (map[string]any, error) {
+	range_ := make(map[string]any)
```

Comment:
**Suggestion**:  Could we rename this variable to be more descriptive or give it a comment explaining what it does?

Reply:
What you don't like mysterious variables that shadow keywords? I'll change it yeah, thanks 👍